tests/test_tpp_backend.py

```diff
@@ -124,3 +129,67 @@ class Cohort:
     assert extract(Cohort, TPPBackend, database) == [
         dict(patient_id=1, admission=date(2020, 12, 12))
     ]
+
+
+@pytest.mark.parametrize(
+    "raw, codes",
+    [
+        ("flim", ["flim"]),
+        ("flim ,flam ,flum", ["flim", "flam", "flum"]),
+        ("flim ||flam ||flum", ["flim", "flam", "flum"]),
+        ("abc ,def ||ghi ,jkl", ["abc", "def", "ghi", "jkl"]),
+        ("ABCX ,XYZ ,OXO", ["ABC", "XYZ", "OXO"]),
+    ],
+    ids=[
+        "returns a single code",
+        "returns multiple space comma separated codes",
+        "returns multiple space double pipe separated codes",
+        "copes with comma pipe combinations",
+        "strips just trailing xs",
+    ],
+)
+@pytest.mark.integration
+def test_hospitalization_table_code_conversion(
+    database, setup_tpp_database, raw, codes
+):
+    setup_tpp_database(
+        *patient(
+            1,
+            "M",
+            registration("2001-01-01", "2026-06-26"),
+            apcs(codes=raw),
+        )
+    )
+
+    class Cohort:
+        code = table("hospitalizations").get("code")
+
+    assert extract(Cohort, TPPBackend, database) == [
+        dict(patient_id=1, code=code) for code in codes
+    ]
+
+
+@pytest.mark.integration
+def test_hospitalization_code_parsing_works_with_filters(database, setup_tpp_database):
+    setup_tpp_database(
+        *patient(
+            1,
+            "X",
+            registration("2001-01-01", "2026-06-26"),
+            apcs(codes="abc"),
+        ),
+        *patient(
+            2,
+            "X",
+            registration("2001-01-01", "2026-06-26"),
+            apcs(codes="xyz"),
+        ),
+    )
+
+    class Cohort:
+        code = table("hospitalizations").filter(code="xyz").get("code")
+
+    assert extract(Cohort, TPPBackend, database) == [
+        dict(patient_id=1, code=None),
+        dict(patient_id=2, code="xyz"),
+    ]
```

Comment:
Do we also need/want a test with an `is_in` `Codelist` filter?  (Not sure it's necessary - I guess if we've tested that this filters ok on a single code, the other codelist tests cover the codelist scenario)

Reply:
What do you think? You know the filtering code much better than me. I'm very happy to add it -- or maybe replace this filter with that one since it seems on the face of it that they would be more likely to interact badly.

Reply:
Yes, maybe make this one a Codelist filter instead